packages/environment/test/develop.js

```diff
@@ -0,0 +1,67 @@
+const assert = require("chai").assert;
+var fs = require("fs-extra");
+var glob = require("glob");
+const { default: Box } = require("@truffle/box");
+const { Environment } = require("@truffle/environment");
+const sinon = require("sinon");
+
+describe("Environment develop", function () {
+  let config;
+
+  const expectedNetwork = {
+    port: 8545,
+    network_id: 1342,
+    gas: 90000
+  };
+
+  const testOptions = {
+    port: 5678,
+    network_id: expectedNetwork.network_id,
+    gas: 7000
+  };
+
+  before("Create a sandbox", async function () {
+    this.timeout(5000);
+    config = await Box.sandbox("default");
+    config.network = "network";
+    config.networks = {
+      network: {
+        port: expectedNetwork.port,
+        network_id: 4321,
+        gas: expectedNetwork.gas
+      }
+    };
+    sinon.stub(Environment, "detect").returns();
+  });
+
+  after("Cleanup tmp files", function (done) {
```

Comment:
I'm not going to push too hard on this but I would prefer to use `async`/`await` syntax in place of the `done` callback.

Reply:
I think we should definitely be avoiding callbacks in new code if there's not a reason we need them...

Reply:
agree, change to async/await